### PR TITLE
Make field and parameter names case-insensitive

### LINKED ISSUE #1: Field/parameter names are not case insensitive
## Describe the bug
Field/parameter names in the framework are case sensitive while in sql they are not

## To Reproduce
Try to add parameter named _param1_ and _PARAM1_. They will be considered as two different ones, while for SQL they are the same.

## Expected behavior
Ensure case-insensitivity of the parameter names.

### PR DESCRIPTION
**Pull request by qwerty44-U**
## Summary
- normalize named query parameters with `Locale.ROOT` on insertion and lookup
- normalize result-set column names from metadata so mixed-case database labels can be accessed case-insensitively
- add regression tests for duplicate mixed-case parameter names and uppercase/mixed-case column metadata

## Verification
- `git diff --check`

Scala tests were not run locally because this environment does not have Java or sbt installed (`java` and `sbt` are not available on PATH).

Fixes #1

## Summary by CodeRabbit

* **Bug Fixes**
  * Column names and named parameters are now handled consistently regardless of letter casing.
  * Lookups now work even when names are entered with different capitalization.
  * Name matching now uses locale-stable lowercasing for more predictable behavior.

* **Tests**
  * Added coverage for case-insensitive column lookup.
  * Added coverage for case-insensitive named parameter handling.

### CODE REVIEW

File: balta/src/main/scala/za/co/absa/db/balta/classes/inner/Params.scala [hunk 1]
```diff
@@ -101,7 +104,7 @@ object Params {
      */
     def add[T: QueryParamType](paramName: String, value: T): NamedParams = {
       val queryValue = implicitly[QueryParamType[T]].toQueryParamValue(value)
-      new NamedParams(items + (paramName -> queryValue)) // TODO https://github.com/AbsaOSS/balta/issues/1
+      new NamedParams(items + (normalizeName(paramName) -> queryValue))
```
[the remark this answers is not in the document]
Comment (lsulak) [reply balta/src/main/scala/za/co/absa/db/balta/classes/inner/Params.scala:107]: Hello @qwerty44-U, I've approved the workflow runs, let's see if the code compiles successfully, including tests. 

However, this one CodeRabbit item is worth considering
